src/game/collision.cpp

```diff
@@ -198,6 +350,92 @@ void CCollision::MoveBox(vec2 *pInoutPos, vec2 *pInoutVel, vec2 Size, float Elas
 					Vel.x *= -Elasticity;
 				}
 			}
+			if (pointer)
+			{
+				if (!TestBox(vec2(Pos.x, Pos.y), Size, COLFLAG_WATER) && TestBox(vec2(NewPos.x, NewPos.y), Size, COLFLAG_WATER))
+				{
+				//	m_pWater->HitWater(NewPos.x, NewPos.y, -pInoutVel->y);
+					//CGameClient::WaterSplash(NewPos.x, NewPos.y, -pInoutVel->y, m_pWater);
+					(*pointer)(NewPos.x, NewPos.y, -pInoutVel->y);
+				}
+				else if (TestBox(vec2(Pos.x, Pos.y), Size, COLFLAG_WATER) && !TestBox(vec2(NewPos.x, NewPos.y), Size, COLFLAG_WATER))
+				{
+					//m_pWater->HitWater(NewPos.x, NewPos.y, pInoutVel->y);
+					(*pointer)(NewPos.x, NewPos.y, pInoutVel->y);
+				}
+			}
+			
+			Pos = NewPos;
+		}
+	}
+	
+	*pInoutPos = Pos;
+	*pInoutVel = Vel;
+}
+void CCollision::MoveWaterBox(vec2* pInoutPos, vec2* pInoutVel, vec2 Size, float Elasticity, bool* pDeath, float Severity) const
```

Comment:
@Stiopa866 this should be really moved into MoveBox since it's the same code, consider refactoring this